src/Url/DrupalGenerator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace IonBazan\ComposerDiff\Url;
+
+use Composer\Package\CompletePackageInterface;
+use Composer\Package\PackageInterface;
+
+class DrupalGenerator extends GitlabGenerator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsPackage(PackageInterface $package)
+    {
+        return 'drupal/core' === $package->getName() || in_array($package->getType(), array('drupal-module', 'drupal-theme')) || parent::supportsPackage($package);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getCompareRef(PackageInterface $package)
+    {
+        if (!$package->isDev()) {
+            return $package->getDistReference();
+        }
+
+        $reference = $package->getSourceReference();
+
+        if (40 === \strlen($reference)) {
+            return \substr($reference, 0, 7);
+        }
+
+        return $reference;
```

Comment:
These cases are not covered by test. Also I think we can call `parent::getCompareRef()` instead when it's a dev package.